src/v2/Apps/Order/Components/SavedAddresses.tsx

```diff
@@ -39,11 +45,56 @@ const defaultAddressIndex = addressList => {
 }
 
 const SavedAddresses: React.FC<SavedAddressesProps> = props => {
+  const [modalDetails, setModalDetails] = useState({
```

Comment:
suggestion: something like this to specify the types?
```ts
useState<{addressModalTitle: string, addressModalAction: AddressModalAction}>({null, null})
```
also `addressModalTitle` can be derived from `AddressModalAction` so it is possible to pass that to the modal component and let the component determine the title baed on the valye of `AddressModalAction`